packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts

```diff
@@ -143,17 +143,20 @@ describeCompat("GC loading from older summaries", "NoCompat", (getTestObjectProv
 		summarizer1.close();
 		await reconnectSummarizerToBeElected(container2);
 
-		// Create a new alias data store so that the GC data changes without changing the GC state of existing data
-		// stores. This is to write the GC tree in summary (instead of handle) which is used for validation.
-		const ds2 = await containerRuntime.createDataStore(TestDataObjectType);
-		const aliasResult = await ds2.trySetAlias("root2");
-		assert.strictEqual(aliasResult, "Success", "Failed to alias data store");
+		// Call summarizeOnDemand. This will trigger the logic that processes summary ack. The ack from the last
+		// summary will cause summarizer2 to dispose because it sees a new ack that is generated by another summarizer.
 		await provider.ensureSynchronized();
-
-		// Summarize - summary3 with the new summarizer. Before it summarizes, it will catch up to latest and so the
-		// reference state of the data stores should be the same as in summary2.
-		// Also, note that while catching up, it will download summary2 and update state from it.
 		await summarizer2.summarizeOnDemand({ reason: "test" }).summarySubmitted;
+
+		// Wait for the container for the above summarizer to be disposed.
+		await timeoutPromise((resolve) => {
+			if (container2.disposed) {
+				resolve();
+			}
+			container2.on("disposed", () => {
+				resolve();
+			});
```

Comment:
nit: you can just pass `resolve` as the 2nd arg
```suggestion
			container2.on("disposed", resolve);
```